### ISSUE TITLE
Feature Request: Better output during builds

### ISSUE THREAD

**Issue by sarahbethfederman** [preconstruct/preconstruct#307]
Heyyy. I'm experiencing pretty long builds atm and it would be helpful to see some sort of output as it builds rather than only showing success or errors at the end after waiting however long (averaging 10+ mins rn 😬 ). Not sure if this is possible or if there's existing rollup output that could be piped through?

**Comment by mitchellhamilton** [comment]
> averaging 10+ mins rn 😬

😱 

---

Yeah, def agree that the output should be better during a build. Slightly worried about the opposite concern of having more output with "Why are you spamming my console with all these things?!?!?!" so for now, I've added an experimental flag `logCompiledFiles` in `@preconstruct/cli@1.1.24` which logs when Babel finishes transforming a file. Add the experimental flag to your root package.json like this:

```json
{
  "preconstruct": {
    "___experimentalFlags_WILL_CHANGE_IN_PATCH": {
      "logCompiledFiles": true
     }
  }
}
```

**Comment by sarahbethfederman** [comment]
Is 10 min a lot? I've got it down to around 7 😅
 
Hm, this is not suuuuper useful as it's running in parallel anyway so it's hard to tell what's being compiled. You're right, it does output a lot. It also gets to the end of the compilation step and hangs, so I'm assuming that's when rollup is doing its thing. Maybe there's a possibility of a progress indicator or some output from rollup

**Comment by mitchellhamilton** [comment]
> Is 10 min a lot? I've got it down to around 7 😅

Yeah, I've personally never seen over a minute. I assume it's a significant amount of code where that sort of time is expected? 

> It also gets to the end of the compilation step and hangs, so I'm assuming that's when rollup is doing its thing.

Hmmm, that's interesting. I've generally found the time rollup spends doing its own thing to rarely be the bottleneck, this made me do some profiling on a sizable TypeScript project though and it looks like generating TS declarations is the biggest cost by a VERY large margin(like ~30 seconds of a ~50 second build) on at least that project so I'll probably look into that at some point, I think there's some good gains that could be made there with TypeScript's incremental mode.

> Maybe there's a possibility of a progress indicator or some output from rollup

Yeah, maybe. There's some complexity and questions to answer in there(We don't know how many modules there are to compile until we've compiled them all so we probably want to store the number of modules from a build to estimate what the future amount will be, what do you output in non-interactive environments like CI and etc., how do other rollup compilation bits and TS declaration generation factor into the progress?) though so I'm probably not gonna look into it anytime soon(PRs welcome though).

**Comment by apuntovanini** [comment]
> Is 10 min a lot? I've got it down to around 7 😅

Our build is also very slow! And hangs at the end of the compilation step when logging is enabled...

**Comment by sarahbethfederman** [comment]
Ours has gone down to 200-300 seconds which could have been due to a lot of factors (updated deps, updated node versions, new CI, new laptop, etc). We do consistently need to set the node memory allocation higher than default else we get an out of memory error both locally and on CI. I'm not sure how to debug or reproduce so I haven't filed an issue but it may be something to keep an eye on. I assume it's lots of parallelization

**Comment by mitchellhamilton** [comment]
Do you have tsconfigs per package? If so, removing them should help - https://github.com/preconstruct/preconstruct/issues/314#issuecomment-685175018
